data_tools/firebase_client_wrapper.py

```diff
@@ -303,10 +303,18 @@ def delete_dataset(dataset_id):
 def _delete_collection(coll_ref, batch_size):
     docs = coll_ref.limit(batch_size).get()
     deleted = 0
+    batch = client.batch()
+    first_id = None
+    last_id = None
     for doc in docs:
-        print(u'Deleting doc {} => {}'.format(doc.id, doc.to_dict()))
-        doc.reference.delete()
+        if first_id is None:
+            first_id = doc.id
+        last_id = doc.id
+
+        batch.delete(doc.reference)
         deleted = deleted + 1
+    batch.commit()
+    print(f"Deleted {deleted} objects ({first_id} to {last_id})")  # Useful because Firebase get()s sort by id
```

Comment:
Is sort by ID guaranteed?

Reply:
Yes. "By default, a query retrieves all documents that satisfy the query in ascending order by document ID" (https://firebase.google.com/docs/firestore/query-data/order-limit-data)